src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.IDictionary.cs

```diff
@@ -224,6 +224,8 @@ public bool Remove(string propertyName)
             return dictionary;
         }
 
+        private bool IsCaseInsensitive => Options?.PropertyNameCaseInsensitive ?? false;
```

Comment:
extracted this bc it's used in the constructor now as well

Reply:
Nit: I think this should preferably be a private static helper method that accepts a `JsonSerializerOptions?` parameter.

Reply:
https://github.com/dotnet/runtime/pull/96486/commits/d9d68236cf6940805381814c6176f16a33ff660c